neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/dialect/impl/RemoteNeo4jEntityQueries.java

```diff
@@ -0,0 +1,334 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.neo4j.remote.dialect.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.hibernate.HibernateException;
+import org.hibernate.ogm.datastore.neo4j.dialect.impl.EntityQueries;
+import org.hibernate.ogm.datastore.neo4j.remote.impl.Neo4jClient;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.ErrorResponse;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.Graph.Node;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.Row;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.Statement;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.StatementResult;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.Statements;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.StatementsResponse;
+import org.hibernate.ogm.dialect.query.spi.ClosableIterator;
+import org.hibernate.ogm.dialect.spi.TupleContext;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
+import org.hibernate.ogm.util.impl.ArrayHelper;
+
+/**
+ * @author Davide D'Alto
+ */
+public class RemoteNeo4jEntityQueries extends EntityQueries {
+
+	private static final ClosableIteratorAdapter<AssociationPropertiesRow> EMPTY_RELATIONSHIPS = new ClosableIteratorAdapter<>(
+			Collections.<AssociationPropertiesRow>emptyList().iterator() );
+
+	private static final ClosableIteratorAdapter<Node> EMPTY_NODES = new ClosableIteratorAdapter<>( Collections.<Node>emptyList().iterator() );
+
+	public RemoteNeo4jEntityQueries(EntityKeyMetadata entityKeyMetadata) {
+		this( entityKeyMetadata, null );
+	}
+
+	public RemoteNeo4jEntityQueries(EntityKeyMetadata entityKeyMetadata, TupleContext tupleContext) {
+		super( entityKeyMetadata, tupleContext );
+	}
+
+	public Node findEntity(Neo4jClient executionEngine, Long transactionId, Object[] columnValues) {
+		Map<String, Object> params = params( columnValues );
+		String query = getFindEntityQuery();
+		List<StatementResult> results = executeQuery( executionEngine, transactionId, query, params );
+		if ( results != null ) {
+			Row row = results.get( 0 ).getData().get( 0 );
```

Comment:
In a single request you can execute multiple queries and each query result can be returned as rows of a  table or as graph elements. All this make things a bit complicated